app/controllers/download_controller.rb

```diff
@@ -65,4 +64,20 @@ def download_user_params
       :register
     )
   end
+
+  def download_from_uri(uri)
+    headers = {}
+    headers['Authorization'] = api_key if api_key.present?
+
+    response = HTTParty.get(uri, headers: headers)
+    if response.code == 200
+      response.body
+    else
+      raise "#{response.status}: #{response.message}"
+    end
+  end
+
+  def api_key
```

Comment:
I found it a bit confusing that is called earlier in the file than where it is defined, can we move it above `download_from_uri`?